### PR TITLE
Make RetainForwardRule public

### PR DESCRIPTION
**Pull request by brianmay**
## Type of change

Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [X] Formatted with `cargo fmt`
- [X] Make an entry to `CHANGELOG.md` if it's relevant to the users of the library. If it's not relevant mention why.

### CODE REVIEW

File: rumqttc/CHANGELOG.md [hunk 1]
```diff
@@ -18,6 +18,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Removed the `Key` enum: users do not need to specify the TLS key variant in the `TlsConfiguration` anymore, this is inferred automatically.
 To update your code simply remove `Key::ECC()` or `Key::RSA()` from the initialization.
 - certificate for client authentication is now optional while using native-tls. `der` & `password` fields are replaced by `client_auth`.
+- Make v5 `RetainForwardRule` public, in order to allow setting it when constructing `Filter` values.
```
Comment (swanandx) [inline rumqttc/CHANGELOG.md:21]:
```suggestion
- `RetainForwardRule` in v5 is now public
```

let's omit the construction of `Filter` values for now as that might be confusing due to `Filters` aren't yet exposed. ( note: not that necessary, feel free to discuss this suggestion haha )

Comment (brianmay) [reply rumqttc/CHANGELOG.md:21]: Huh? `Filter` is exposed, and the reason I am making this PR.

Comment (swanandx) [reply rumqttc/CHANGELOG.md:21]: My bad 😬 I thought the PR of setting v5 subscription options was making it public 😅

**Review by swanandx — APPROVED** [review]
Thank you for the PR!

**Pull request merged**